### ISSUE TITLE
No event on disconnect when making a server stream request

### ISSUE THREAD

**Issue by rokn** [grpc/grpc-node#2578]
### Problem description
When using `makeServerStreamRequest` I'm expecting to receive an error if the connection has been interrupted(like turning of the machine wifi). Instead there is no events emitted.

### Reproduction steps
I've prepared a simple POC of the problem which can be easily tested in this repo: https://github.com/rokn/grpc-server-stream-disconnect-poc

### Environment
 - MacOS M1
 - Node: 16.16
 - nvm
 - "@grpc/grpc-js": "1.9.3"

### Additional context
It could be I'm missing some documentation but I haven't found a way to do this

**Comment by murgatroid99** [comment]
It usually should detect if a connection is dropped, but sometimes it just doesn't happen. I recommend enabling keepalives using the channel options `grpc.keepalive_time_ms` (controls how often pings are sent) and `grpc.keepalive_timeout_ms` (controls how long the client waits for a ping response before determining that a connection no longer exists).

**Comment by dangerous1990** [comment]
I have same issue, I have using consul for name resolver, resolver watch has changed, the client stream request is nothing happened in 30s. I'm not found channel close func with load balance or resolver interface

**Comment by rokn** [comment]
Ok with the keepalive time param it's correctly notifying for an UNAVAILABLE service but it would be perfect if there is some other way to more quickly detect a network error as the gRPC server I'm communicating with has a 90 second rate limit on keepalive pings

**Comment by dangerous1990** [comment]
I have another solution, the server stream listen system signal (SIGINT SIGTERM) and close the stream , the client stream will   
callback immediately.

**Comment by rokn** [comment]
I'm not exactly sure what you are proposing but if it's closing the stream from the server side that's not the use case I have. I don't have control over the server and really I need a solution when the user machine is left without internet, like a wifi disconnect or something of that sorts

**Comment by dangerous1990** [comment]
yes is not a good idea.  the right way how to  observe the channel closed or not work. my case is i have know the address not work ,i have unref the channel, but the on request stream not work immediately
